parser/src/lexer.rs

```diff
@@ -1292,6 +1309,7 @@ where
 
     // Helper function to emit a lexed token to the queue of tokens.
     fn emit(&mut self, spanned: Spanned) {
+        self.last_emitted = Some(spanned.0.clone());
```

Comment:
This seems a bit expensive as we'll be cloning every single token for the entire source code. Can we maybe look at the pending token instead?

Reply:
Yeah, that's very expensive because it even has to clone the heap allocations of the tokens. 

Some optimisations: Only store the token kind (see ruff-python-ast `TokenKind`) which is only a u8. 

Track the context explicitly: 

https://github.com/mozilla/sweet.js/wiki/design and an implementation of it https://github.com/swc-project/swc/blob/026101b71e942ad2d7ff906cb68bf46345d77712/crates/swc_ecma_parser/src/lexer/state.rs#L756-L797

Reply:
> In your case it seems all you really care about is whether you've seen an `equal` token.

Yes, I think having a boolean stating "Was the last token an `Equal`?" seems like the simplest option. I'll explore a bit around lexer context and state after this as it's pretty interesting :)